### PR TITLE
Add validations for server and api keys to client

### PR DESCRIPTION
**Pull request by Lytol**
Fixes known issue for server URLs without scheme. Also gives better error messaging around missing server and API keys.

### CODE REVIEW

File: R/rscpages.R [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-#' rscpages provides UI components to help publishers provide curated organization of content on RStudio Connect.
+#' rscpages provides UI components to help publishers provide curated
+#' organization of content on RStudio Connect.
 #'
 #' @section Connection:
-#' Set `CONNECT_SERVER` and `CONNECT_API_KEY` environment variables and [connect()].
-#' Fetch a `tibble` listing server content with [content()].
-#' use the [by_owner()] and [by_tags()] filter helpers to narrow your search.
+#' Set `CONNECT_SERVER` and `CONNECT_API_KEY` environment variables and
+#' [connect()]. Fetch a `tibble` listing server content with [content()]. use
```
Comment (edavidaja) [inline R/rscpages.R:8]: Probably meant to be upper case on the next line?

File: DESCRIPTION [hunk 2]
```diff
@@ -31,8 +31,11 @@ Imports:
     digest,
     crosstalk,
     reactR,
-    purrr
+    purrr,
+    withr (>= 2.4.2)
```
Comment (edavidaja) [inline DESCRIPTION:35]: Does this need to be in `Imports` if it's only being used for tests?

File: DESCRIPTION [hunk 3]
```diff
@@ -31,8 +31,11 @@ Imports:
...
 URL:
     https://rstudio.github.io/rscpages,
     https://github.com/rstudio/rscpages
 VignetteBuilder: knitr
+Remotes:
```
Comment (edavidaja) [inline DESCRIPTION:40]: I don't think the DESCRIPTION can contain this [when it goes to CRAN](https://cran.r-project.org/web/packages/devtools/vignettes/dependencies.html#cran-submission)--do we need something in particular from the dev version?

**Review by edavidaja — CHANGES_REQUESTED** [review]
Small dependencies question but looking good otherwise.

File: DESCRIPTION (hunk 2 above)
Comment (Lytol) [reply DESCRIPTION:35]: Good point.  I just used usethis::use_dev_package() and this is where it put it.  Curious.

File: R/rscpages.R (hunk 1 above)
Comment (Lytol) [reply R/rscpages.R]: Fixed

**Review by edavidaja — APPROVED** [review]
👍🏾

**Pull request merged**